server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -567,103 +523,76 @@ private void becomeLeader()
-
-        final List<Pair<? extends CoordinatorRunnable, Duration>> coordinatorRunnables = Lists.newArrayList();
+
+      metadataSegmentManager.start();
+      metadataRuleManager.start();
+      serviceAnnouncer.announce(self);
+      final int startingLeaderCounter = coordLeaderSelector.localTerm();
+
+      final List<Pair<? extends CoordinatorRunnable, Duration>> coordinatorRunnables = Lists.newArrayList();
+      coordinatorRunnables.add(
+          Pair.of(
+              new CoordinatorHistoricalManagerRunnable(startingLeaderCounter),
+              config.getCoordinatorPeriod()
+          )
+      );
+      if (indexingServiceClient != null) {
         coordinatorRunnables.add(
             Pair.of(
-                new CoordinatorHistoricalManagerRunnable(startingLeaderCounter),
-                config.getCoordinatorPeriod()
+                new CoordinatorIndexingServiceRunnable(
+                    makeIndexingServiceHelpers(),
+                    startingLeaderCounter
+                ),
+                config.getCoordinatorIndexingPeriod()
             )
         );
-        if (indexingServiceClient != null) {
-          coordinatorRunnables.add(
-              Pair.of(
-                  new CoordinatorIndexingServiceRunnable(
-                      makeIndexingServiceHelpers(),
-                      startingLeaderCounter
-                  ),
-                  config.getCoordinatorIndexingPeriod()
-              )
-          );
-        }
+      }
 
-        for (final Pair<? extends CoordinatorRunnable, Duration> coordinatorRunnable : coordinatorRunnables) {
-          ScheduledExecutors.scheduleWithFixedDelay(
-              exec,
-              config.getCoordinatorStartDelay(),
-              coordinatorRunnable.rhs,
-              new Callable<ScheduledExecutors.Signal>()
-              {
-                private final CoordinatorRunnable theRunnable = coordinatorRunnable.lhs;
+      for (final Pair<? extends CoordinatorRunnable, Duration> coordinatorRunnable : coordinatorRunnables) {
+        ScheduledExecutors.scheduleWithFixedDelay(
+            exec,
+            config.getCoordinatorStartDelay(),
+            coordinatorRunnable.rhs,
+            new Callable<ScheduledExecutors.Signal>()
+            {
+              private final CoordinatorRunnable theRunnable = coordinatorRunnable.lhs;
 
-                @Override
-                public ScheduledExecutors.Signal call()
-                {
-                  if (leader && startingLeaderCounter == leaderCounter) {
-                    theRunnable.run();
-                  }
-                  if (leader && startingLeaderCounter == leaderCounter) { // (We might no longer be leader)
-                    return ScheduledExecutors.Signal.REPEAT;
-                  } else {
-                    return ScheduledExecutors.Signal.STOP;
-                  }
+              @Override
+              public ScheduledExecutors.Signal call()
```

Comment:
what is the intended behavior if leadership is lost during this call?

Reply:
Yes, this also means if lookup management is taking a long time (minutes) then this is going to continue doing lookup management until it either completes or errors out, then it will give up leadership.

In such a scenario it is possible for the new leader to also do lookup management at the same time.

Is there a way to mitigate such a scenario?

Reply:
well it is mitigated in the sense that even if multiple coordinators are doing lookup management , nothing bad happens... in the worst case, downstream nodes would receive request to load same lookup multiple times and those requests would be ignored.